golem-cli/src/command/worker.rs

```diff
@@ -402,44 +460,94 @@ impl<ComponentRef: clap::Args, WorkerRef: clap::Args> WorkerSubcommand<Component
                 idempotency_key,
                 function,
                 parameters,
+                connect,
+                connect_options,
             } => {
                 let (worker_uri, project_ref) = worker_ref.split();
                 let project_id = projects.resolve_id_or_default_opt(project_ref).await?;
-                service
-                    .invoke_and_await(
-                        format,
-                        worker_uri,
-                        idempotency_key,
-                        function,
-                        parameters.parameters,
-                        parameters.wave,
-                        project_id,
-                    )
-                    .await
+                if connect {
+                    let worker_uri_clone = worker_uri.clone();
+                    let project_id_clone = project_id.clone();
+                    let service_clone = service.clone();
+                    let connect_handle = spawn(async move {
+                        let _ = service_clone
+                            .connect(worker_uri_clone, project_id_clone, connect_options, format)
+                            .await;
+                    });
+                    let result = service
+                        .invoke_and_await(
+                            format,
+                            worker_uri,
+                            idempotency_key,
+                            function,
+                            parameters.parameters,
+                            parameters.wave,
+                            project_id,
+                        )
+                        .await;
+
+                    connect_handle.abort();
```

Comment:
is this and unconditional connect /  abort concurrently to the invoke? i'm not sure i understand the error handling and side effect of this

Reply:
we stream the worker output concurrently to the invoke, and when the invoke returns we abort it.

Reply:
what confuses me: is there a guarantee that connect succeed and streaming is started before invoke starts (and ends)?

Reply:
and also i'm not sure what is the diff between how it handled here (invoke and await) and in invoke